lighthouse-core/gather/gatherers/dobetterweb/links.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+function getPageLinks() {
+  return new Promise((resolve, reject) => {
+    const linkList = [...document.querySelectorAll('a')]
```

Comment:
OTOH, I kinda like the idea of keeping the links gatherer general, in case another audit can make use of it.